src/tsickle.ts

```diff
@@ -557,16 +561,57 @@ class Annotator extends ClosureRewriter {
   private templateSpanStackCount = 0;
   private polymerBehaviorStackCount = 0;
 
+  /**
+   * The set of module symbols forward declared in the local namespace (with goog.forwarDeclare).
+   *
+   * Symbols not imported must be declared, which is done by adding forward declares to
+   * `extraImports` below.
+   */
+  private forwardDeclaredModules = new Set<ts.Symbol>();
+  private extraDeclares = '';
+
   constructor(
       typeChecker: ts.TypeChecker, file: ts.SourceFile, host: AnnotatorHost,
       private tsHost?: ts.ModuleResolutionHost, private tsOpts?: ts.CompilerOptions,
       sourceMapper?: SourceMapper) {
     super(typeChecker, file, host, sourceMapper);
   }
 
-  annotate() {
+  annotate(): {output: string, diagnostics: ts.Diagnostic[]} {
     this.visit(this.file);
-    return this.getOutput();
+    return this.getOutput(this.extraDeclares);
+  }
+
+  ensureSymbolDeclared(sym: ts.Symbol) {
+    // TODO(martinprobst): it's unclear when a symbol wouldn't have a declaration, maybe just for
+    // some builtins (e.g. Symbol)?
+    if (!sym.declarations || sym.declarations.length === 0) return;
+    // A symbol declared in this file does not need to be imported.
+    if (sym.declarations.some(d => d.getSourceFile() === this.file)) return;
+    // Find an exported declaration.
+    // Because tsickle runs with the --declaration flag, all types referenced from exported types
+    // must be exported, too, so there must either be some declaration that is exported, or the
+    // symbol is actually a global declaration (declared in a script file, not a module).
+    const decl = sym.declarations.find(
+        d => hasModifierFlag(d, /* Export | Default */ ts.ModifierFlags.ExportDefault));
+    if (!decl) return;
+    // Exclude symbols declared in `declare global {...}` blocks, they are global and don't need
+    // imports.
+    let current: ts.Node|undefined = decl;
+    while (current) {
+      if (current.flags & ts.NodeFlags.GlobalAugmentation) return;
```

Comment:
Is it possible there's some other declaration that satisfies your requirements? (E.g. that .find picked the wrong one above?) Might be worth refactoring so you call .some() over all declarations and do this filtering within the .some?

Reply:
Done.